platform/security/servlet/security-servlet-session-expiry/src/main/java/org/codice/ddf/security/servlet/expiry/SessionManagementService.java

```diff
@@ -0,0 +1,156 @@
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.Response;
+
+import org.apache.cxf.ws.security.tokenstore.SecurityToken;
+import org.codice.ddf.security.handler.api.SAMLAuthenticationToken;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.security.SecurityConstants;
+import ddf.security.Subject;
+import ddf.security.assertion.SecurityAssertion;
+import ddf.security.assertion.impl.SecurityAssertionImpl;
+import ddf.security.common.SecurityTokenHolder;
+import ddf.security.service.SecurityManager;
+import ddf.security.service.SecurityServiceException;
+
+@Path("/")
+public class SessionManagementService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(SessionManagementService.class);
+
+    private SecurityManager securityManager;
+
+    private Clock clock = Clock.systemUTC();
+
+    @GET
+    @Path("/expiry")
+    public Response getExpiry(@Context HttpServletRequest request)
+            throws ServletException, IOException {
+        HttpSession session = request.getSession(false);
+        long timeLeft = 0;
+        if (session != null) {
+            Object securityToken = session.getAttribute(SecurityConstants.SAML_ASSERTION);
+            if (securityToken instanceof SecurityTokenHolder) {
+                List<SecurityAssertionImpl> values =
+                        ((SecurityTokenHolder) securityToken).getRealmTokenMap()
+                                .values()
+                                .stream()
+                                .map(SecurityAssertionImpl::new)
+                                .collect(Collectors.toList());
+                values.sort((o1, o2) -> {
+                    long l = o1.getNotOnOrAfter()
+                            .getTime() - o2.getNotOnOrAfter()
+                            .getTime();
+                    if (l > 0) {
+                        return 1;
+                    } else if (l < 0) {
+                        return -1;
+                    } else {
+                        return 0;
+                    }
+                });
+                timeLeft = values.isEmpty() ?
+                        0 :
+                        values.get(0)
+                                .getNotOnOrAfter()
+                                .getTime();
+            }
+        }
+        return Response.ok(new ByteArrayInputStream(Long.toString(Math.max(
+                timeLeft - clock.millis(), 0))
+                .getBytes(StandardCharsets.UTF_8)))
+                .build();
+    }
+
+    @GET
+    @Path("/renew")
+    public Response getRenewal(@Context HttpServletRequest request) {
+        HttpSession session = request.getSession(false);
+        final Response[] response = new Response[1];
```

Comment:
❓ Why are you using an array of responses when you always just use a single index?

Reply:
Lambda below requires effectively final reference.